.azure-pipelines/generation-pipeline.yml

```diff
@@ -49,94 +51,227 @@ resources:
-  typeScriptBetaBranch: 'pipelinebuild/beta/$(Build.BuildId)'
-
-steps:
-- pwsh: |
-    git config --global user.email "GraphTooling@service.microsoft.com"
-    git config --global user.name "Microsoft Graph DevX Tooling"
-  displayName: 'Git: set user config'
-
-- checkout: msgraph-metadata
-  displayName: checkout metadata
-  fetchDepth: 1
-  persistCredentials: true
-
-- checkout: self
-  displayName: checkout generator
-  fetchDepth: 1
-  submodules: recursive
-  persistCredentials: true
-
-- checkout: msgraph-sdk-dotnet
-  displayName: checkout dotnet
-  fetchDepth: 1
-  persistCredentials: true
-
-- checkout: msgraph-beta-sdk-dotnet
-  displayName: checkout beta dotnet
-  fetchDepth: 1
-  persistCredentials: true
-
-- checkout: msgraph-sdk-php
-  displayName: checkout php
-  fetchDepth: 1
-  persistCredentials: true
-
-- checkout: msgraph-typescript-typings
-  displayName: checkout typescript
-  fetchDepth: 1
-  persistCredentials: true
-
-- checkout: msgraph-sdk-objc-models
-  displayName: checkout objc
-  fetchDepth: 1
-  persistCredentials: true
-
-- checkout: msgraph-sdk-java
-  displayName: checkout java
-  fetchDepth: 1
-  persistCredentials: true
-
-- checkout: msgraph-beta-sdk-java
-  displayName: checkout beta java
-  fetchDepth: 1
-  persistCredentials: true
-
-- template: common-templates/use-dotnet-sdk.yml
-
-- task: DotNetCoreCLI@2
-  displayName: 'Build Typewriter'
-  inputs:
-    command: 'build'
-    projects: '$(Build.SourcesDirectory)/MSGraph-SDK-Code-Generator/src/Typewriter/Typewriter.csproj'
-    arguments: '--configuration $(buildConfiguration)'
-
-- template: generation-templates/dotnet-v1.yml
-- template: generation-templates/dotnet-beta.yml
-- template: generation-templates/php.yml
-- template: generation-templates/typescript.yml
-- template: generation-templates/objc.yml
-- template: generation-templates/java-v1.yml
-- template: generation-templates/java-beta.yml
\ No newline at end of file
+  typeScriptBetaBranch: ${{ parameters.typeScriptBetaBranchPrefix }}/$(Build.BuildId)
+
+stages:
+- stage: stage_build_and_publish_typewriter
+  dependsOn: [] # remove the implicit dependency to any previous stage
+  jobs:
+  - job: build_and_publish_typewriter
+    steps:
+    - template: generation-templates/build-and-publish-typewriter.yml
+
+- stage: stage_v1_metadata
+  dependsOn: stage_build_and_publish_typewriter
+  condition: succeeded('stage_build_and_publish_typewriter')
+  jobs:
+  - job: v1_metadata
+    steps:
+    - template: generation-templates/checkout-metadata.yml
+    - pwsh: |
+        Write-Host "Success!"
+
+- stage: stage_beta_metadata
+  dependsOn: stage_build_and_publish_typewriter
+  condition: succeeded('stage_build_and_publish_typewriter')
+  jobs:
+  - job: beta_metadata
+    steps:
+    - template: generation-templates/checkout-metadata.yml
+    - pwsh: |
+        Write-Host "Success!"
+
+- stage: stage_csharp_v1
+  dependsOn:
+  - stage_v1_metadata
```

Comment:
shouldn't this stage depend on both the metadata and the typewriter ones?

Reply:
it depends on typewriter step by the transitive property:

typewriter -> metadata -> generation.

Reply:
But should the metadata one depend on typewriter at the first place? Whatever happens in the metadata stages could run in parralels of the typewriter one

Reply:
metadata cleaning step consumes typewriter executable (TransformWithDocs).